cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -809,6 +791,12 @@ func (s *Supervisor) loadInitialMergedConfig() error {
 		return fmt.Errorf("could not compose initial merged config: %w", err)
 	}
 
+	// write the initial merged config to disk
+	cfg := s.mergedConfig.Load().(string)
+	if err := os.WriteFile(agentConfigFilePath, []byte(cfg), 0600); err != nil {
+		s.logger.Error("Failed to write agent config.", zap.Error(err))
+	}
+
```

Comment:
This code was needed for the test to work; I wanted to start the collector with the persisted remote config, but it looked like that wasn't working and that we never run the persisted config. 

These lines write the config to disk so that that previous config is run instead of always starting with the bootstrap config.

Reply:
>This code was needed for the test to work

Just to clarify, it's not only needed for the test, right? Also, should we be writing this through the persisted state?

Reply:
Right, this is not just needed for tests, but it also is needed for the supervisor to use the last received config on startup, which is the intended behavior.

I'm not sure I follow on persisted state; The config file here is constructed from the persisted RemoteConfig message received from the OpAMP server, and I think just having that persisted is sufficient.

Reply:
My bad, I forgot that the storage directory and persistent state aren't both managed by the persistent state. I was asking about it because I noticed the config file doesn't appear to reside in the same directory that the persisted state goes into, but we can handle that separately.